packages/rest-typings/src/v1/oauthapps.ts

```diff
@@ -1,73 +1,6 @@
 import type { IOAuthApps, IUser } from '@rocket.chat/core-typings';
-import Ajv from 'ajv';
 
-const ajv = new Ajv({
-	coerceTypes: true,
-});
-
-export type OauthAppsGetParams = { clientId: string } | { appId: string } | { _id: string };
-
-const oauthAppsGetParamsSchema = {
-	oneOf: [
-		{
-			type: 'object',
-			properties: {
-				clientId: {
-					type: 'string',
-				},
-			},
-			required: ['clientId'],
-			additionalProperties: false,
-		},
-		{
-			type: 'object',
-			properties: {
-				appId: {
-					type: 'string',
-				},
-			},
-			required: ['appId'],
-			additionalProperties: false,
-		},
-		{
-			type: 'object',
-			properties: {
-				appId: {
-					type: 'string',
-				},
-			},
-			required: ['_id'],
-			additionalProperties: false,
-		},
-	],
-};
-
-export const isOauthAppsGetParams = ajv.compile<OauthAppsGetParams>(oauthAppsGetParamsSchema);
-
-export type OauthAppsAddParams = {
-	name: string;
-	active: boolean;
-	redirectUri: string;
-};
-
-const OauthAppsAddParamsSchema = {
-	type: 'object',
-	properties: {
-		name: {
-			type: 'string',
-		},
-		active: {
-			type: 'boolean',
-		},
-		redirectUri: {
-			type: 'string',
-		},
-	},
-	required: ['name', 'active', 'redirectUri'],
-	additionalProperties: false,
-};
-
-export const isOauthAppsAddParams = ajv.compile<OauthAppsAddParams>(OauthAppsAddParamsSchema);
+import type { OauthAppsGetParams, UpdateOAuthAppParams, OauthAppsAddParams, DeleteOAuthAppParams } from '..';
```

Comment:
This import is a circular reference. The current bundler might be smart enough to overcome it, but others may not. External consumers of this package can import from barrel exports, but internally is better to reference directly the modules implementing the desired structure.
```suggestion
import type { OauthAppsGetParams } from './oauthapps/OauthAppsGetParams';
import type { UpdateOAuthAppParams } from './oauthapps/UpdateOAuthAppParams';
import type { OauthAppsAddParams } from './oauthapps/OauthAppsAddParams';
import type { DeleteOAuthAppParams } from './oauthapps/DeleteOAuthAppParams';
```